### PR TITLE
Added UATP card type

### PR DESCRIPTION
**Pull request by subpx**
https://en.wikipedia.org/wiki/Payment_card_number
https://en.wikipedia.org/wiki/Universal_Air_Travel_Plan
https://www.getnewidentity.com/uatp-debit-card.php

### CODE REVIEW

**Comment by bendrucker** [comment]
Hi, can you find any first-party documentation? I don't consider Wikipedia or some random card generator to be valid sources.

**Comment by subpx** [comment]
Knew that would be the case. 
I wish it was that easy to get docs.

**Comment by bendrucker** [comment]
Ok. I'm happy to review this with _any_ first party or even apparently reliable source. Even documentation from a gateway is acceptable. Just not Wikipedia, card validators, or anyone else who's re-sourcing generic info.

**Comment by subpx** [comment]
Here is the most credible stuff I could find. 

https://www.cybersource.com/developers/other_resources/quick_references/test_cc_numbers/
https://docs.adyen.com/developers/test-cards/test-card-numbers
https://baymard.com/checkout-usability/credit-card-patterns
https://www.paypalobjects.com/webstatic/en_US/developer/docs/pdf/ppspec_uatpauth.pdf

**Comment by bendrucker** [comment]
Paypal's docs could be workable, please see if you can excerpt a section that makes explicit reference to a BIN range.

**Comment by toemouse** [comment]
Thanks for your work, Im just going to use my own fork.
I recommend you adding UATP if you ever find any documentation you find acceptable.

**Comment by bendrucker** [comment]
No need for a fork! You can construct a type yourself and add it to the API:

https://github.com/bendrucker/creditcards-types#api

I put in some time last months to make that functionality available across this lib and [creditcards](github.com/bendrucker/creditcards).

**Comment by bendrucker** [comment]
```
Character length and limits: An alphanumeric
string of 15 digits. The first 4 digits are the
PayPal reserved bin 1112;
```

From PayPal's document, page 26. I can't seem to find any other sources referencing that sequence but if that's what PayPal says that's what I think we should go with now. If you make that update I'm happy to merge.

I'm sending an email to UATP's general address to see if they have something official that they'd be willing to share. If they don't respond I'd like someone else to confirm that UATP actually has the whole `^1` range.

File: types/uatp.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+var Type = require('../type')
+
+module.exports = Type({
+  name: 'UATP',
+  digits: 15,
+  pattern: /^1\d{14}$/,
+  eagerPattern: /^1/,
+  groupPattern: /(1\d{2})(\d{1,4})?(\d{1,4})?(\d{1,4})?/
```
Comment (bendrucker) [inline types/uatp.js]: Google search indicates this is not right:

https://www.google.com/search?q=uatp+card&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiQr67Y6oPZAhUO1mMKHfGMAY0Q_AUICigB&biw=2127&bih=1321#imgrc=nY11YBAnx6I-aM:

File: types/uatp.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+'use strict'
+
+var Type = require('../type')
+
+module.exports = Type({
+  name: 'UATP',
+  digits: 15,
+  pattern: /^1\d{14}$/,
+  eagerPattern: /^1/,
+  groupPattern: /(1\d{3})(\d{1,5})?(\d{1,6})?/
```
Comment (bendrucker) [inline types/uatp.js]: Google search indicates this is not right:

https://www.google.com/search?q=uatp+card&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiQr67Y6oPZAhUO1mMKHfGMAY0Q_AUICigB&biw=2127&bih=1321#imgrc=nY11YBAnx6I-aM:

**Comment by bendrucker** [comment]
If both of you have a UATP card or handle them and can personally verify that they have the whole 1 range this thread is good enough documentation for me.

Saw that you fixed the group pattern, thanks for that!

**Comment by bendrucker** [comment]
Tests do need to be updated to match

File: types/uatp.js (hunk 2 above)
Comment (bendrucker) [inline types/uatp.js]: This should just be `\d{4}` no need to match on the 1 in this regex

**Comment by bendrucker** [comment]
This is good to merge now, I'd appreciate two comments confirming that you've seen a valid UATP number falling outside PayPal's documented BIN.

**Comment by subpx** [comment]
Yes I can confirm this. I don't have full card numbers but have the first 6 digits.
All start with 1 and are 15 digits in length. No other obvious pattern.

**Comment by bendrucker** [comment]
Hey sorry for letting this hang. I just got a notification since you're making a bunch of syntax changes. I won't be able to merge with either of the last two commits you made. Let me know if you're still interested in landing this.

**Comment by subpx** [comment]
Changed back sorry, didn't realise this hadn't been merged yet. Was just mucking around with something totally unrelated.

**Comment by bendrucker** [comment]
No problemo, merged!

**Pull request merged**